tock/tock/middleware.py

```diff
@@ -0,0 +1,27 @@
+from django.conf import settings
+from django.contrib import auth
+from datetime import datetime, timedelta
+
+
+class AutoLogout():
+    def process_request(self, request):
+        # Check if user is logged in at all
+        if not request.user.is_authenticated():
+            return
```

Comment:
I would consider either 

```
if request.user and request.user.is_authenticated():
    do stuff
```
or
```
if not request.user or request.user.is_anonymous():
    do stuff
````
I may be over-guarding, but I'd be watching for any weird cases where request.user wasn't as expected.